### PR TITLE
Updates @types 

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] update some `@types`
- [x] use patch instead of kb-custom for react-native overrides
- [x] use `useNativeDriver` to match what rn 62 wants
- [x] use `private` in touched files
- [x] use Return type instead of nodejs.timer

### CODE REVIEW

File: shared/common-adapters/toast.native.tsx [hunk 1]
```diff
@@ -41,6 +42,7 @@ const Toast = (props: Props) => {
         duration: 100,
         easing: NativeEasing.linear,
         toValue: 1,
+        useNativeDriver: false,
```
Comment (buoyad) [inline shared/common-adapters/toast.native.tsx]: why these changes?

Comment (chrisnojima) [reply shared/common-adapters/toast.native.tsx]: https://github.com/facebook/react-native/commit/5876052615f4858ed5fc32fa3da9b64695974238

Comment (buoyad) [reply shared/common-adapters/toast.native.tsx]: ah ok
